src/screens/CameraScreen.js

```diff
@@ -1,26 +1,296 @@
-const CameraScreen = (props) => {
-  return (
-    <View style={styles.screen}>
-      <Text>Camera Screen</Text>
-      <Button
-        title="Posting"
-        onPress={() => {
-          props.navigation.navigate({ routeName: "Posting" });
-        }}
-      />
-    </View>
-  );
-};
+import React, { Component } from "react";
+import { View, Text, SafeAreaView, StatusBar, StyleSheet } from "react-native";
+import { MaterialIcons } from "@expo/vector-icons";
+import { Camera } from "expo-camera";
+import * as Permissions from "expo-permissions";
+import { Video } from "expo-av";
+import { TouchableOpacity } from "react-native-gesture-handler";
+import Colors from "../constants/Colors";
+
+export default class CameraScreen extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      hasCameraPermissions: null,
+      type: Camera.Constants.Type.back,
+      flash: Camera.Constants.FlashMode.off,
+      videoTaken: false,
+      cameraIsRecording: false,
+      video: null,
+    };
+  }
+
+  componentDidMount = async () => {
+    const camera = await Permissions.askAsync(Permissions.CAMERA);
+    this.setState({
+      hasCameraPermissions: camera.status === "granted",
+    });
+  };
+
+  _changeType = () => {
+    this.setState((prevState) => {
+      if (prevState.type === Camera.Constants.Type.back) {
+        return { type: Camera.Constants.Type.front };
+      } else {
+        return { type: Camera.Constants.Type.back };
+      }
+    });
+  };
+
+  _changeFlash = () => {
+    this.setState((prevState) => {
+      if (prevState.flash === Camera.Constants.FlashMode.off) {
+        return { flash: Camera.Constants.FlashMode.on };
+      } else if (prevState.flash === Camera.Constants.FlashMode.on) {
+        return { flash: Camera.Constants.FlashMode.auto };
+      } else if (prevState.flash === Camera.Constants.FlashMode.auto) {
+        return { flash: Camera.Constants.FlashMode.off };
+      }
+    });
+  };
+
+  _takeVideo = async () => {
+    if (this.camera) {
+      const takenVideo = await this.camera.recordAsync();
+      this.setState({ video: takenVideo.uri, videoTaken: true });
+    }
+  };
+
+  _rejectVideo = () => {
+    this.setState({
+      video: null,
+      videoTaken: false,
+    });
+  };
+
+  _cancelCamera = async () => {
+    const { navigation } = this.props;
+    navigation.goBack(null);
+  };
+
+  _goToLibrary = async () => {
+    const {
+      navigation: { navigate },
+    } = this.props;
+    navigate("Library");
+  };
+
+  _approveVideo = async () => {
+    const { video } = this.state;
+    const {
+      navigation: { navigate },
+    } = this.props;
+    navigate("Upload", { url: video });
+    this.setState({
+      video: null,
+      videoTaken: false,
+    });
+  };
+
+  render() {
+    const {
+      hasCameraPermissions,
+      type,
+      flash,
+      videoTaken,
+      video,
+      cameraIsRecording,
+    } = this.state;
+
+    if (hasCameraPermissions === null) {
```

Comment:
Can this object be undefined too? Wondering if it's better to use `!hasCameraPermission` to account for undefined too.

Reply:
Disregard the above comment, we can't use `!hasCameraPermission` here since that would also succeed for `false`. But let me know if we want to handle the undefined case though.

Reply:
I checked the code. I already set the hasCameraPermissions null. So if the user did deny the Permissions, it will be set to false and it will display like no camera access, please check settings to access. If the user approves, it will get to true and it won't ask again on their device. I don't think it can be undefined but if you want to do it, you can add it and show me how. I'm not expert haha
![Screen Shot 2021-04-18 at 1 20 43 PM](https://user-images.githubusercontent.com/55858439/115159610-e2e9e180-a048-11eb-91f5-c66fa05d2c58.png)

Reply:
Oh no need, thanks for looking into this Shane!